worker/uniter/uniter.go

```diff
@@ -438,12 +433,20 @@ func (u *Uniter) loop(unitTag names.UnitTag) (err error) {
 		}
 	}
 
+	var rebootDetected bool
+
 	for {
 		if err = restartWatcher(); err != nil {
 			err = errors.Annotate(err, "(re)starting watcher")
 			break
 		}
 
+		if u.modelType == model.IAAS {
+			if rebootDetected, err = u.rebootQuerier.Query(unitTag); err != nil {
```

Comment:
Nice catch. I missed that rebootDetected was memoized outside the loop.

FYI,  an alternative way to fix this that doesn't query the FS each time you enter the loop (guess that's probably quite rare so it doesn't really matter) would be to create the reboot resolver outside the loop. As you can see from its implementation, it reports the initial value that gets passed to its c-tor once and then always returns false.